internal/start/start.go

```diff
@@ -531,9 +531,22 @@ EOF
 			fmt.Sprintf("GOTRUE_RATE_LIMIT_WEB3=%v", utils.Config.Auth.RateLimit.Web3),
 		}
 
-		// Since signing key is validated by ResolveJWKS, simply read the key file.
 		if keys, err := afero.ReadFile(fsys, utils.Config.Auth.SigningKeysPath); err == nil && len(keys) > 0 {
 			env = append(env, "GOTRUE_JWT_KEYS="+string(keys))
+			algSet := map[string]bool{"HS256": true}
+			for _, key := range utils.Config.Auth.SigningKeys {
+				switch key.Algorithm {
+				case config.AlgRS256:
+					algSet["RS256"] = true
+				case config.AlgES256:
+					algSet["ES256"] = true
+				}
+			}
+			algorithms := make([]string, 0, len(algSet))
+			for alg := range algSet {
+				algorithms = append(algorithms, alg)
+			}
+			env = append(env, "GOTRUE_JWT_VALID_METHODS="+strings.Join(algorithms, ","))
```

Comment:
```suggestion
			if len(utils.Config.Auth.SigningKeys) > 0 {
				env = append(env, "GOTRUE_JWT_VALID_METHODS=RS256,ES256")
			}
```

I think it's simpler to list all supported algorithms.

Also `HS256` only applies to legacy JWT keys which has been deprecated. So we don't need to include it in the list of valid algs.